File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.rb
def largest_integer(nums, k)
  cnt = Array.new(51, 0)

  nums.each do |x|
    cnt[x] += 1
  end

  return nums.max if k == nums.length

  if k == 1
    ans = -1

    nums.each do |x|
      ans = [ans, x].max if cnt[x] == 1
    end

    return ans
  end

  a = cnt[nums[0]] == 1 ? nums[0] : -1
  b = cnt[nums[-1]] == 1 ? nums[-1] : -1

  [a, b].max
end

File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.php
class Solution
{
    function largestInteger($nums, $k)
    {
        $cnt = array_fill(0, 51, 0);

        foreach ($nums as $x)
        {
            $cnt[$x]++;
        }

        if ($k == count($nums))
        {
            return max($nums);
        }

        if ($k == 1)
        {
            $ans = -1;

            foreach ($nums as $x)
            {
                if ($cnt[$x] == 1)
                {
                    $ans = max($ans, $x);
                }
            }

            return $ans;
        }

        $a = $cnt[$nums[0]] == 1 ? $nums[0] : -1;
        $b = $cnt[$nums[count($nums) - 1]] == 1
            ? $nums[count($nums) - 1]
            : -1;

        return max($a, $b);
    }
}